### PR TITLE
Use vs-threading analyzers to enforce Async naming convention

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: src/PerformanceTests/Tests/Enabled/CSharp_CA1416.cs [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+#pragma warning disable VSTHRD200 // Use "Async" suffix for async methods
```
Comment (Evangelink) [inline src/PerformanceTests/Tests/Enabled/CSharp_CA1416.cs:3]: I am wondering if it would be worth to have a specific `.editorconfig` in the `PerformanceTests` section.

File: src/PublicApiAnalyzers/UnitTests/DeclarePublicApiAnalyzerTests.cs [hunk 2]
```diff
@@ -1685,6 +1685,7 @@ public C(int value)
             await VerifyCSharpAdditionalFileFixAsync(source, shippedText, unshippedText, fixedUnshippedText);
         }
 
+        [Fact]
```
Comment (Evangelink) [inline src/PublicApiAnalyzers/UnitTests/DeclarePublicApiAnalyzerTests.cs:1688]: Nice!

Shall it be moved to a separate PR?

File: src/Roslyn.Diagnostics.Analyzers/Core/RelaxTestNamingSuppressor.cs [hunk 3]
```diff
@@ -27,7 +27,9 @@ public sealed class RelaxTestNamingSuppressor : DiagnosticSuppressor
 
         public override void ReportSuppressions(SuppressionAnalysisContext context)
         {
-            if (context.Compilation.GetOrCreateTypeByMetadataName(WellKnownTypeNames.XunitFactAttribute) is not { } factAttribute)
+            context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.XunitFactAttribute, out var factAttribute);
+            context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.BenchmarkDotNetAttributesBenchmarkAttribute, out var benchmarkAttribute);
```
Comment (Evangelink) [inline src/Roslyn.Diagnostics.Analyzers/Core/RelaxTestNamingSuppressor.cs:31]: Shall it be moved to a separate PR?

File: src/PublicApiAnalyzers/UnitTests/DeclarePublicApiAnalyzerTests.cs (hunk 2 above)
Comment (sharwell) [reply src/PublicApiAnalyzers/UnitTests/DeclarePublicApiAnalyzerTests.cs:1688]: No, this fixes the VSTHRD200 violation on the method.

File: src/PerformanceTests/Tests/Enabled/CSharp_CA1416.cs (hunk 1 above)
Comment (sharwell) [reply src/PerformanceTests/Tests/Enabled/CSharp_CA1416.cs:3]: ➡️ These will become unnecessary after Roslyn.Diagnostics.Analyzers is updated and the new RelaxNamingSuppressor rules take effect.

File: src/Roslyn.Diagnostics.Analyzers/Core/RelaxTestNamingSuppressor.cs (hunk 3 above)
Comment (sharwell) [reply src/Roslyn.Diagnostics.Analyzers/Core/RelaxTestNamingSuppressor.cs:31]: ➡️ This one _could_ be moved to a different PR, but this non-production package is the analyzers specifically created for dealing with scenarios like this.

**Review by Evangelink — APPROVED** [review]

File: src/Roslyn.Diagnostics.Analyzers/Core/RelaxTestNamingSuppressor.cs (hunk 3 above)
Comment (sharwell) [reply src/Roslyn.Diagnostics.Analyzers/Core/RelaxTestNamingSuppressor.cs:31]: @jmarolf do you have an order preference here?

Comment (jmarolf) [reply src/Roslyn.Diagnostics.Analyzers/Core/RelaxTestNamingSuppressor.cs:31]: I am fine with this change being a part of this PR

**Review by jmarolf — APPROVED** [review]
:shipit:

File: src/Roslyn.Diagnostics.Analyzers/Core/RelaxTestNamingSuppressor.cs (hunk 3 above)
Comment (sharwell) [reply src/Roslyn.Diagnostics.Analyzers/Core/RelaxTestNamingSuppressor.cs:31]: Thanks, I'll revert 07c9509 as part of the analyzer upgrade after the signed build is published.
